### PR TITLE
incorporates promises, sponsorships in dwwi decision

### PR DESCRIPTION
**Pull request by mekarpeles**
### Description

Closes #2267 

We still need some way @hornc to make it so partners don't give us duplicate books of those we've sponsored

### CODE REVIEW

File: openlibrary/core/sponsorships.py [hunk 1]
```diff
@@ -91,6 +91,39 @@ def get_sponsorships_by_contact_id(contact_id, isbn=None):
...
+        return False, availability
+
+    # We don't have any of these work's editions available to borrow
+    # Let's confirm this edition hasn't already been sponsored or promised
+    params = {
+        'search_field': 'isbn',
+        'include_promises': 'true',  # include promises and sponsored books
+        'search_id': isbn
+    }
+    domain = 'https://www-judec.archive.org'  # replace w/ config param once me
```
Comment (mekarpeles) [inline openlibrary/core/sponsorships.py]: this is blocking; we'll switch to the production url once it's merged

File: openlibrary/utils/isbn.py [hunk 2]
```diff
@@ -47,6 +47,10 @@ def isbn_10_to_isbn_13(isbn_10):
     isbn_13 = '978' + isbn_10[:-1]
     return isbn_13 + check_digit_13(isbn_13)
 
+def to_isbn_13(isbn):
```
Comment (hornc) [inline openlibrary/utils/isbn.py]: We should add some tests for this.
There probably should be some tests passing an isbn13 to `test_isbn_10_to_isbn_13()` and isbn10 to `test_isbn_13_to_isbn_10()`

`utils.normalise_isbn()` should be used more where hyphens are being stripped using custom code in other places.

**Comment by hornc** [comment]
@mekarpeles Creating a data item with the sponsored ISBN 13s under the archive.org collection: https://archive.org/details/protodonationitems  would be the way to achieve that.   You'll have to tell me at exactly which point in the process the ISBN should be written there.

Currently I'm thinking a single Sponsorship program item should exist under that collection, and it can be written to using the ia client each time an ISBN is added?

File: static/css/components/buttonCta.less [hunk 3]
```diff
@@ -21,12 +21,45 @@ a.cta-btn {
...
+  }
+
+  &--sponsor {
+    background: @white-blue;
+    border: 2px solid @link-blue;
+  }
+
+  &--sponsor:link {
+      color: @link-blue;
   }
```
Comment (tabshaikh) [inline static/css/components/buttonCta.less]: @mekarpeles the conflict in this file 
``` 
=======
  &--preview, &--preview:link, &--preview:visited {
    background-color: @white;
    border: 2px solid @primary-blue;
    color: @primary-blue;
  }
>>>>>>> master-sponsorship
``` 
Needs to be kept. These changes were made by @shivam to fix the bug

File: static/css/components/buttonCta.less [hunk 4]
```diff
@@ -21,12 +21,45 @@ a.cta-btn {
...
+  &--sponsor:visited {
+    color: @link-blue;
+  }
+  &--sponsor:hover {
+    border: 2px solid @link-blue;
+    background-color: @link-blue;
+    color: @white;
+    text-decoration: none;
+  }
+
```
Comment (tabshaikh) [inline static/css/components/buttonCta.less]: The conflict here can be solved by
``` 
<<<<<<< sponsorship-dwwi-api
=======
```

**Comment by mekarpeles** [comment]
This should now be ready to go.

**Pull request merged**